TESTING.md

```diff
@@ -63,12 +63,47 @@ tests are finished. The command below will give you a shell after all tests are
 
 ```
 # cd keylime-tests
-# tmt run -vvv prepare discover provision -h virtual -i Fedora-35 execute login finish
+# tmt run -vvv prepare discover provision -h virtual -i Fedora-35 -c system execute login finish
 ```
 
 You can use it to inspect test logs or even modify test sources and run your tests
 manually following the method described above. For this purpose you can find test sources under `/var/tmp/tmt/run-*/packit-ci/discover/default/tests/`.
 
+### Running multi-host tests
+
+`tmt` cannot schedule multi-host tests yet. However, we can use `tmt` to do the necessary setup and then execute the test manually.
+
+For `Multihost/basic-attestation` test we need at least 3 test systems that
+will be started using `tmt`.
+
+Let's start with opening 3 terminals on your workstation.
+
+On terminal 1:
+```
+$ git clone https://github.com/RedHat-SP-Security/keylime-tests.git
+```
+
+On all terminals:
+```
+$ cd keylime-tests
+$ tmt run -vvv prepare discover -h fmf -t 'configure_tpm_emulator' -t 'install_upstream_keylime' -t 'Multihost/basic-attestation' provision -h virtual -i Fedora-35 execute login finish
+```
+
+Multihost test won't be run properly but at least tmt will install all test requires and do the setup.
+
+Once we have an interactive shell on every test system we can proceed with test execution.
+
+On every test system:
+```
+# cd /var/tmp/tmt/run-*/plans/keylime-tests-github-ci/discover/default/tests/Multihost/basic-attestation/
+# ## find out hostname or IP of each test system, e.g. with "hostname -I"
```

Comment:
Hmm.  I wonder if I did something wrong here -- all 3 interactive shells show the same hostname and IP(v4), apparently:
`interactive shell #1`
```
[root@testcloud tree]# hostname -I
172.17.2.15 10.0.2.15 fec0::5054:ff:fe54:b399 fec0::c1b2:38:2b6e:4682
```

`interactive shell #2`
```
[root@testcloud tree]# hostname -I
172.17.2.15 10.0.2.15 fec0::5054:ff:fe80:4302 fec0::21b2:1242:25a1:f8cd
```

`interactive shell #3`
```
[root@testcloud tree]# hostname -I
172.17.2.15 10.0.2.15 fec0::5054:ff:fe03:1dc9 fec0::175:71:5c26:eae1
```

Reply:
And it seems I was using the old tmt command line... let me retest with the extra `-c system`

Reply:
And apparently it does; at least I see already a different address for each system:
```
progress: booting...
ip: 192.168.124.163
```
```
progress: booting...
ip: 192.168.124.116
```
```
progress: booting...
ip: 192.168.124.162
```
OK, I guess I reached the point of your last comment, that it still won't work because we don't have a sync mechanism.

Reply:
Thank you. I am working on a custom sync mechanism so it can be used by folks outside RH.